ixmp/core.py

```diff
@@ -1174,19 +1198,24 @@ def clone(self, model=None, scenario=None, annotation=None,
                 ' release, please use `scenario`')
             scenario = kwargs.pop('scen')
 
-        first_model_year = first_model_year or 0
+        if keep_solution and first_model_year is not None:
+            raise ValueError('Use `keep_solution=False` when cloning with '
+                             '`first_model_year`!')
+
+        if platform is not None and not keep_solution:
+            raise ValueError('Cloning across platforms is only possible '
```

Comment:
Because the Java backend is a mess and implementing all features would even take more time. So the current approach is the first least-bad option:
 - if you are cloning in the same platform, you can discard the solution on the fly or shift the first model year
 - you can create an exact duplicate in a difference platform